addons/editor/functions/fnc_deepCopy.sqf

```diff
@@ -0,0 +1,154 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Returns deep copy data for the given objects.
+ *
+ * Arguments:
+ * 0: Objects <ARRAY>
+ *
+ * Return Value:
+ * Object Data <ARRAY>
+ *
+ * Example:
+ * [_objects] call zen_editor_fnc_deepCopy
+ *
+ * Public: No
+ */
+
+params ["_objects"];
+
+// Filter destroyed objects and get the vehicle for mounted infantry
+_objects = _objects select {alive _x && {isNull isVehicleCargo _x}} apply {vehicle _x};
+_objects = _objects arrayIntersect _objects;
+
+// Get the world position where the cursor is pointing
+// Object positions are copied relative to the cursor
+private _centerPos = if (visibleMap) then {
+    private _ctrlMap = findDisplay IDD_RSCDISPLAYCURATOR displayCtrl IDC_RSCDISPLAYCURATOR_MAINMAP;
+    private _pos2D = _ctrlMap ctrlMapScreenToWorld getMousePosition;
+    _pos2D + [0]
+} else {
+    screenToWorld getMousePosition
+};
+
+private _units = [];
+private _vehicles = [];
+private _statics = [];
+
+private _groups = [];
+private _indexedGroups = [];
+
+private _fnc_serializeGroup = {
+    params ["_unit"];
+
+    private _group = group _unit;
+    private _index = _indexedGroups find _group;
+
+    if (_index == -1) then {
+        _index = _indexedGroups pushBack _group;
+
+        _groups pushBack [side _group, formation _group, behaviour leader _group, combatMode _group, speedMode _group];
+    };
+
+    _index
+};
```

Comment:
Maybe extract this whole feature to a new component and make each of these functions separate? Would make this main code much cleaner.

Reply:
While I do agree that it would nicer for these individual serializing functions to be in a separate file, I think the use of the local variables such as `_centerPos` and `_indexedGroups` might become cumbersome as these would then need to be passed to the respective functions rather than relying on them being defined.